initial work on requirements chapter (#24)
mqtt-varroa/varroa-documentation: LaTeX/Parts/Chapter7.tex

\chapter{Requirements}
The following sections list the requirements and their respective priorities.

> In addition, it is indicated whether and how they are currently implemented.

Reviewer: ```suggestion
Additionally it is indicated whether and how they are currently implemented.
```

\section{Non-Functional Requirements}

\subsection{Transparency (Must Have High)}\label{sec:Transparency}
\emph{Varroa has to be comprehensible for the user.}

This requirement is fulfilled.
Varroa implements expressive logging with configurable levels and outputs.

> This enables rich insight in the execution process of a Varroa test.

Reviewer: ```suggestion
This enables a rich insight in the execution process of a Varroa test.
```
Don't know whether this is an improvement, but the sentence feels incomplete.

\subsection{10.000.000 MQTT Clients (Must Have High)} 
\emph{Varroa has to be able to generate a large amount of clients.}

> The fulfilment of this requirement is not verified yet.

Reviewer: This sentence feels very incomplete.

Reviewer 2: It just says that we could not verify this yet. It is okay imho.

\subsection{Scalability (Must Have)} 
\emph{Varroa should scale vertically with relatively low scaling costs.}

This requirement is fulfilled.
To do so Varroa is organised as a distributed System and can easily be extended with new instances running as Varroa Agents.

\subsection{Determinism (Must Have)} 

> \emph{Varroa has to work in deterministic ways, meaning it should produce the same result for a Scenario every time.}

Reviewer: ```suggestion
\emph{Varroa has to work in deterministic ways, meaning it should produce the same result for the same Scenario every time.}
```

This requirement is fulfilled.

> To ensure this determinism several measures are taken in the distribution and execution of a Scenario (see chapter \ref{sec:Architecture}).

Reviewer: ```suggestion
To guarantee this determinism several measures are taken, ensuring deterministic distribution and execution of a Scenario (see chapter \ref{sec:Architecture}).
```
Not sure about the semicolon.

\subsection{Distributed (Must Have Low)} 
\emph{Varroa is a distributed System.}

This requirement is fulfilled.

> Varroa is organized as a distributed System, it is composed of a Commander and multiple Agents that can run on different machines. 

Reviewer: ```suggestion
Varroa is organized as a distributed System.
It is composed of a Commander and multiple Agents.
These components can be run on separate machines. 
```

 
\subsection{Usabillity (Very Important)} 
\emph{Varroa has to be easily usable.}

This requirement is fulfilled.
The user can easily define custom Scenarios as XML files and execute them (see chapter \ref{sec:ScenarioConcept} and \ref{sec:Execution}).

\subsection{Code Quality (Important)} 
\emph{Varroa's code quality should be very high.}

This requirement is fulfilled.

> To ensure high code quality several measures are taken.

Reviewer: ```suggestion
To ensure the high code quality standards, several measures were taken.
```
Sentence still is pretty generic, we should have a further look at this.

Reviewer 2: Because of this there are the following sentences.

> For example the consequent use of nullability annotiations and a broad test code coverage.

Reviewer: ```suggestion
An example for these measures is the consequent use of nullability as well as broad test code coverage. 
```

Also before new code could be added to the master branch it was reviewed by another team member.

\subsection{Stability (Important)} 
\emph{Varroa has to run in a stable manner.}

This requirement is fulfilled.

> By the use of integration tests and general unit tests stability is ensured.

Reviewer: ```suggestion
Stability is ensured by the use of integration tests and general unit tests.
```

\subsection{Resource efficiency (Important)} 
\emph{Varroa has to use the available computation and memory resources efficiently.}
%TODO how was this verified.

\subsection{User / Developer Guide (Somewhat Important)} 
\emph{Varroa needs a User / Developer Guide.}

> This requirement is fulfilled. This documentation educates the user on defining a custom scenario and after that executing it on his Varroa Distributed System (\ref{sec:ScenarioConcept} and \ref{sec:Execution}).

Reviewer: ```suggestion
This requirement is fulfilled. 
The documentation educates the user on the definition of a custom scenario aswell as the exectution of it on her Varroa Distributed System (\ref{sec:ScenarioConcept} and \ref{sec:Execution}).
```
Georg said we gender everything feminine.

> This requirement is fulfilled. This documentation educates the user on defining a custom scenario and after that executing it on his Varroa Distributed System (\ref{sec:ScenarioConcept} and \ref{sec:Execution}).

Reviewer 2: ```suggestion
This requirement is fulfilled.
The documentation educates the user on the definition of a custom scenario aswell as the exectution of it on his Varroa Distributed System (\ref{sec:ScenarioConcept} and \ref{sec:Execution}).
```
